projects/valtimo/process-link/src/lib/components/select-url/select-url.component.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {BehaviorSubject, combineLatest, filter, map, Observable, of, Subscription, switchMap, tap} from 'rxjs';
+import {
+  ProcessLinkButtonService,
+  ProcessLinkStateService,
+  ProcessLinkService,
+} from '../../services';
+import {distinctUntilChanged, take} from 'rxjs/operators';
+import {
+  AbstractControl,
+  AsyncValidatorFn,
+  FormControl,
+  FormGroup,
+  ValidationErrors,
+  ValidatorFn,
+  Validators
+} from '@angular/forms';
+import {URLProcessLinkUpdateRequestDto} from '../../models';
+import {UrlValidatorService} from '../../services/url-validator.service';
+import {UrlResolverService} from '../../services/url-resolver.service';
+
+@Component({
+  selector: 'valtimo-select-url',
+  templateUrl: './select-url.component.html',
+})
+export class SelectUrlComponent implements OnInit, OnDestroy {
+  public readonly saving$ = this.stateService.saving$;
+  private readonly _variables$ = new BehaviorSubject<Map<string, string>>(null);
+
+  private _subscriptions = new Subscription();
+
+  public urlForm: FormGroup;
+
+  constructor(
+    private readonly stateService: ProcessLinkStateService,
+    private readonly buttonService: ProcessLinkButtonService,
+    private readonly processLinkService: ProcessLinkService,
+    private readonly urlValidatorService: UrlValidatorService,
+    private readonly urlResolverService: UrlResolverService
+  ) {
+  }
+
+  public ngOnInit(): void {
+    this.openBackButtonSubscription();
+    this.openSaveButtonSubscription();
+
+    this.urlForm = new FormGroup({
+      url: new FormControl('', Validators.required, this.urlValidatorService.urlValidator(this._variables$))
+    });
```

Comment:
This can be added directly where you declared the urlForm, no need for instantiation in OnInit